File: iroha-android-sample/src/main/java/jp/co/soramitsu/iroha/android/sample/main/receive/ReceiveFragment.java
package jp.co.soramitsu.iroha.android.sample.main.receive;

import android.databinding.DataBindingUtil;
import android.graphics.Bitmap;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.jakewharton.rxbinding2.widget.RxTextView;

import javax.inject.Inject;

import jp.co.soramitsu.iroha.android.sample.R;
import jp.co.soramitsu.iroha.android.sample.SampleApplication;
import jp.co.soramitsu.iroha.android.sample.databinding.FragmentReceiveBinding;
import jp.co.soramitsu.iroha.android.sample.main.MainActivity;

public class ReceiveFragment extends Fragment implements ReceiveView {
    private FragmentReceiveBinding binding;

    @Inject
    ReceivePresenter presenter;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_receive, container, false);
        SampleApplication.instance.getApplicationComponent().inject(this);

        presenter.setFragment(this);
        presenter.generateQR(binding.amount.getText().toString());
        RxTextView.textChangeEvents(binding.amount)
                .subscribe(text -> presenter.generateQR(text.text().toString()),
                        this::didError);
        return binding.getRoot();
    }

    @Override
    public void didGenerateSuccess(Bitmap bitmap) {
        binding.qrCodeImageView.setImageBitmap(bitmap);
    }

    @Override
    public void didError(Throwable error) {
        ((MainActivity) getActivity()).showError(error);
    }

    @Override
    public void onStop() {
        super.onStop();
        presenter.onStop();
    }
}

File: app/src/main/java/jp/co/soramitsu/irohaandroid/presentation/main/receive/ReceiveFragment.kt
package jp.co.soramitsu.irohaandroid.presentation.main.receive

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import com.jakewharton.rxbinding2.widget.RxTextView
import jp.co.soramitsu.irohaandroid.IrohaSampleApp

import jp.co.soramitsu.irohaandroid.R
import jp.co.soramitsu.irohaandroid.presentation.base.BaseFragment
import jp.co.soramitsu.irohaandroid.presentation.main.MainViewModel
import kotlinx.android.synthetic.main.fragment_receive.amount
import kotlinx.android.synthetic.main.fragment_receive.qr_code_image_view

class ReceiveFragment : BaseFragment<MainViewModel>() {

    companion object {
        fun newInstance(): ReceiveFragment {
            return ReceiveFragment()
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_receive, container, false)
    }

    override fun initViews() {
        RxTextView.textChanges(amount)
            .subscribe { viewModel.getQrImage(it.toString()) }
    }

    override fun inject() {
        (activity!!.application as IrohaSampleApp)
            .appComponent
            .mainComponentBuilder()
            .withActivity(activity as AppCompatActivity)
            .build()
            .inject(this)
    }

    override fun subscribe(viewModel: MainViewModel) {
        observe(viewModel.qrBitmapLiveData, Observer {
            qr_code_image_view.setImageBitmap(it)
        })
    }
}
